### PR TITLE
Small cleanups to span parent id functionality

### PR DESCRIPTION
**Pull request by manugoyal**
- Use Enum instead of StrEnum due to lack of availability in older pythons

- Add `export` to `NoopSpan` (and thus allow returning None/undefined from the method).

### CODE REVIEW

**Review by ankrgyl — COMMENTED** [review]
Should we just return "" instead of `undefined` from the no-op span, so that users don't have to deal with handling undefined everywhere now?

**Comment by manugoyal** [comment]
> Should we just return "" instead of `undefined` from the no-op span, so that users don't have to deal with handling undefined everywhere now?

Good idea. Done.

File: js/src/logger.ts [hunk 1]
```diff
@@ -1911,33 +1920,7 @@ export function traced<IsAsyncFlush extends boolean = false, R = void>(
   callback: (span: Span) => R,
   args?: StartSpanArgs & SetCurrentArg & AsyncFlushArg<IsAsyncFlush>
 ): PromiseUnless<IsAsyncFlush, R> {
-  const { span, isLogger } = ((): { span: Span; isLogger: boolean } => {
```
Comment (manugoyal) [inline js/src/logger.ts:1914]: Unrelated cleanup: I noticed that the lower-level `startSpan` function was not looking at the `parent` arg, so I factored out the bit of logic that considers it and call it in both places.

**Review by ankrgyl — APPROVED** [review]
